Add non-root url path support

hcl opened the pull request:
As issue #26 described, this pull request has added the support for non-root path URL, which should work behind a reverse proxy.

Here are the changes in the PR:
1. Add a flag `-base` as suggested in #26. 
2. Change the URL with root prefix with relative prefix in the front-end pages and api requests.
3. Change the route definition for base path support.
4. Add redirection handler when access non-root path without `/` suffix.

I have made some simple test and it should work well.

nkanaev commented:
@hcl thanks for your contribution! the patch looks perfect, and I'm happy to merge it.

I have one suggestion - can we handle basepaths differently? I think parsing urls during runtime, directly in `ServeHTTP` would be more elegant and should eliminate extra goroutine setup step & extra handler. The preprocessed/trimmed url then could be passed to `getRoute` (which requires only path string, not the whole Request object).

hcl commented:
Thanks for your kindly suggestions. The initial commit is indeed lack of consideration of future development.  I have moved non-root url route handling into `ServeHTTP` function. I also changed the function argument of the `getRoute` from `http.Request` to `string`.

nkanaev commented:
@hcl thanks!
